python-builder/README.md

```diff
@@ -0,0 +1,120 @@
+
+# How To Use This Image
+
+## Create a `Dockerfile` in your Python package's project
+
+```dockerfile
+FROM ramseyinhouse/python-builder:2.7
+
+COPY . /code
+
+RUN setup \
+    && validate
+```
+
+Put this file in the root of your project, next to `requirements.txt` or `Pipfile`.
+
+This image includes the following default scripts for your convenience. They live at `/usr/local/bin`. If you want to override them, `/code/bin` is set to the front of the `PATH` to make that easy.
+- setup
+    - `pip install pipenv`
+    - `pipenv install --dev --deploy` - install package dependencies
+        - Will use Pipfile if it exists in the package's directory, or will default to requirements.txt.
+- validate  - `pipenv run pytest`
+- build - Does a `pipenv run python setup.py sdist bdist_wheel`, and leaves the `.tar.gz` (source) & `.whl` (binary) files in the `/code/dist/` folder inside the container.
+- publish - automatically publishes your package to the repositories specified in `$UPLOAD_REPOS` ([see this section](#publishing-your-package)).
+
+The default command is the validate script. This means you can build your docker image and run your tests within the container like so:
+
+```console
+$ docker build -t your-package .
+$ docker run -it --rm your-package
+```
+
+## Building your package
+
+You can build your package using the provided command:
+
+```console
+$ docker run -it --rm -v "$PWD"/dist:/code/dist your-package build
+```
+
+## Publishing your package
+
+If you'd like to have your package automatically published to several repositories, you can do the following:
```

Comment:
We need to be careful about putting anything out publicly. Right now, anything that goes out Open Source actually needs prior Dave approval. The Dave.

Reply:
Instead of "be careful", I should say "follow the process". @philipdowner can help with that.